crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2589,6 +2640,25 @@ impl<'db> TypeInferenceBuilder<'db> {
                 }
             }
 
+            (
+                Type::BooleanLiteral(b1),
+                Type::BooleanLiteral(b2),
+                ruff_python_ast::Operator::BitOr,
+            ) => match (b1, b2) {
+                (true, true) => Type::BooleanLiteral(true),
+                (true, false) => Type::BooleanLiteral(true),
+                (false, true) => Type::BooleanLiteral(true),
+                (false, false) => Type::BooleanLiteral(false),
+            },
```

Comment:
This can be simplified:
```suggestion
            ) => Type::BooleanLiteral(b1 | b2),
```